File: p3/deepspace/src/deepspace/ShieldBooster.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package deepspace;

/**
 *
 * @author Sergio
 */
public class ShieldBooster {
    
    private String name;
    private float boost;
    private int uses;
    
    ShieldBooster(String name, float boost, int uses){
        this.name = name;
        this.boost = boost;
        this.uses = uses;
    }
    
    ShieldBooster( ShieldBooster s){
        this.name = s.name;
        this.boost = s.boost;
        this.uses = s.uses;
    }
    
    float getBoost(){
        return this.boost;
    }
    
    int getUses(){
        return this.uses;
    }
    
    float useIt(){
        float ret;
        if(uses>0){
            --uses; 
            ret = boost;
        }
        else
            ret = 1.0f;
        
        return ret;
    }
    
    ShieldToUI getUIversion(){
        return new ShieldToUI(this);
    }
}

File: p3/Deepspace/ShieldBooster.rb
#encoding:utf-8
require_relative "ShieldToUI"
module Deepspace
    class ShieldBooster
        def initialize(_name,_boost,_uses)
            @name = _name
            @boost = _boost
            @uses = _uses
        end 
        
        def self.newCopy(s)
            ShieldBooster.new(s.name,s.boost,s.uses)
        end 

        attr_reader:name,:boost,:uses
        
        def useIt()
            if (@uses > 0)
                @uses = @uses - 1
                @boost
            else 
                return 1.0
            end
        end 
        
        def to_s()
            return "Boost: #{@boost}, Uses: #{@uses}"
        end

        def getUIversion()
			Deepspace::ShieldToUI.new(self)
		end        
    end 
    #a = ShieldBooster.new("prueba",1,2)
    #puts a.to_s
end 